### PR TITLE
Add RzBinResource type and use it to print resources

### LINKED ISSUE #1733: Convert `bin_resources()` from `librz/core/cbin.c` into the proper API
Currently it just prints. This whole chunk of functions should be split in two main pieces: one to extract all information and another to print it:
- `rz_core_bin_resources_print()`
- `rz_bin_resources_info()`

Note, that the actual API function to get the information about versions could be moved to `librz/bin/resources.c`, for example, since it doesn't require anything `RzCore`-specific. The printing functions we can keep in `librz/core/cbin.c` as before.

```c
RZ_API bool rz_core_bin_resources_print(RzCore *core, RzBinFile *bf, RzCmdStateOutput *state) {
	rz_return_val_if_fail(core && state, false);

	// TODO: add rz_bin_object_get_resources and switch to table + json output
	switch (state->mode) {
	case RZ_OUTPUT_MODE_STANDARD:
		bin_resources(core, NULL, RZ_MODE_PRINT);
		break;
	case RZ_OUTPUT_MODE_JSON:
		bin_resources(core, state->d.pj, RZ_MODE_JSON);
		break;
	default:
		rz_warn_if_reached();
		break;
	}
	return true;
}

static int bin_resources(RzCore *r, PJ *pj, int mode) {
	const RzBinInfo *info = rz_bin_get_info(r->bin);
	if (!info || !info->rclass) {
		return false;
	}
	if (!strncmp("pe", info->rclass, 2)) {
		bin_pe_resources(r, pj, mode);
	} else {
		bin_no_resources(r, pj, mode);
	}
	return true;
}
```

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Add `RzBinResource`  data type, similarly to RzBinMem, RzBinReloc and others, so that we can work on the data in an easier way also from Cutter and provide a better API in general.

**Test plan**

Use `iR` on both binaries providing resources and some which don't.

**Closing issues**

Closes #1733

### CODE REVIEW

File: librz/bin/bin.c [hunk 1]
```diff
@@ -179,6 +179,16 @@ RZ_API void rz_bin_import_free(RzBinImport *imp) {
 	}
 }
 
+RZ_API void rz_bin_resource_free(RzBinResource *res) {
+	if (res) {
```
Comment (XVilka) [inline librz/bin/bin.c]: Reverse the condition

File: librz/core/cmd_descs/cmd_info.yaml [hunk 2]
```diff
@@ -309,6 +309,7 @@ commands:
     type: RZ_CMD_DESC_TYPE_ARGV_STATE
     modes:
       - RZ_OUTPUT_MODE_JSON
+      - RZ_OUTPUT_MODE_TABLE
```
Comment (XVilka) [inline librz/core/cmd_descs/cmd_info.yaml:312]: Please add a test for this case then.

**Review by XVilka — APPROVED** [review]
Looks good.

**Comment by ret2libc** [comment]
Few other comments. Not everything is mandatory to fix now, but the rz_pdb.h is. The rest we can probably improve in a second (smaller) PR. Thanks a lot for your work!

**Comment by XVilka** [comment]
> Thanks a lot for your work!

It's always good to appreciate your own work ;)

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
You seem to forgot to include the header in the table:
```
[XX] db/cmd/cmd_i iRt
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iRt bins/pe/standard.exe
-- stdout
--- expected
+++ actual
@@ -1,5 +1,7 @@
-0     1576 ICON       0x00401180 0x1628 ICON       Thu Jan  1 01:00:00 1970 UTC+1
-1     10   STRING     0x00402a10 0x4c   STRING     Thu Jan  1 01:00:00 1970 UTC+1
-2     788  GROUP_ICON 0x004027b0 0x14   GROUP_ICON Thu Jan  1 01:00:00 1970 UTC+1
-3     1    VERSION    0x004027d0 0x1c4  VERSION    Thu Jan  1 01:00:00 1970 UTC+1
-4     1    MANIFEST   0x004029a0 0x4a   MANIFEST   Thu Jan  1 01:00:00 1970 UTC+1
+index name type       vaddr      size   lang         timestamp                    
+----------------------------------------------------------------------------------
+0     1576 ICON       0x00401180 0x1628 LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+1     10   STRING     0x00402a10 0x4c   LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+2     788  GROUP_ICON 0x004027b0 0x14   LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+3     1    VERSION    0x004027d0 0x1c4  LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+4     1    MANIFEST   0x004029a0 0x4a   LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC

```

**Comment by ret2libc** [comment]
> > Thanks a lot for your work!
> 
> It's always good to appreciate your own work ;)

LOOOOOL it was meant for another Pr, not sure how it arrived here. Damn vscode-github plugin.

**Comment by XVilka** [comment]
Still:
```
[XX] db/cmd/cmd_i iRt
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iRt:quiet bins/pe/standard.exe
-- stdout
--- expected
+++ actual
@@ -1,5 +1,5 @@
-0     1576 ICON       0x00401180 0x1628 LANG_NEUTRAL Thu Jan  1 01:00:00 1970 UTC+1
-1     10   STRING     0x00402a10 0x4c   LANG_NEUTRAL Thu Jan  1 01:00:00 1970 UTC+1
-2     788  GROUP_ICON 0x004027b0 0x14   LANG_NEUTRAL Thu Jan  1 01:00:00 1970 UTC+1
-3     1    VERSION    0x004027d0 0x1c4  LANG_NEUTRAL Thu Jan  1 01:00:00 1970 UTC+1
-4     1    MANIFEST   0x004029a0 0x4a   LANG_NEUTRAL Thu Jan  1 01:00:00 1970 UTC+1
+0     1576 ICON       0x00401180 0x1628 LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+1     10   STRING     0x00402a10 0x4c   LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+2     788  GROUP_ICON 0x004027b0 0x14   LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+3     1    VERSION    0x004027d0 0x1c4  LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
+4     1    MANIFEST   0x004029a0 0x4a   LANG_NEUTRAL Thu Jan  1 00:00:00 1970 UTC
```

**Pull request merged**